src/features/metadata/metadataSubscriber.ts

```diff
@@ -1,16 +1,32 @@
 import { store } from "../../app/store";
-import { getActiveTokensLocalStorageKey } from "./metadataApi";
+import {
+  getActiveTokensLocalStorageKey,
+  getTransactionsLocalStorageKey } from "./metadataApi";
+import { SubmittedTransaction } from '../transactions/transactionsSlice';
+
 
 export const subscribeToSavedTokenChangesForLocalStoragePersisting = () => {
   const cache: {
     [chainId: number]: string[];
   } = {};
+  const transactionCache: {
+    [chainId: number]: SubmittedTransaction[];
+  } = {}
   store.subscribe(() => {
-    const { wallet, metadata } = store.getState();
+    const { wallet, metadata, transactions } = store.getState();
     if (!wallet.connected) return;
+    // persist all transactions to localStorage
```

Comment:
Discussion point for later - we should probably have a limit on the number of transactions we store (even if it's quite high!)
There should also be the ability to clear this. 

I think we should address both of these as a separate PR though, because otherwise this one will become a monster!